util/util.go

```diff
@@ -112,11 +112,6 @@ func ResolveName(name string, firstRegistry string, sc *types.SystemContext, sto
 		logrus.Debugf("unable to read configured registries to complete %q: %v", name, err)
 		registries = []string{}
 	}
```

Comment:
When do we reference the insecure registries list?  Rather than appending the insecure registries, should we just use only them when DockerInsecureSkipTLSVerify is set?  I.e. just do 
registries = unverifiedRegistries 
at line 117

Reply:
If you want to search a registry then you add it to the search able list, if you want to treat a registry as insecure, then you put it in the insecure list, If you want them in both then you put them in both. 

@vrotherberg is working on a redesigned config, but this is what we have now.

Reply:
@TomSweeneyRedHat I believe we have another bug on mishandling of insecure registiries, which might be what you are looking at.